src/world/action/sneaky_parasol.c

```diff
@@ -1,52 +1,368 @@
+        playerStatus->flags &= ~0x80000000;
+        mem_clear(&D_802B6E80, 0x24);
+        disable_player_static_collisions();
+        tempUnk_1C = &parasolStruct->unk_1C;
+        playerStatus->decorationList = 0;
+        playerStatus->unk_C2 = 0;
+        playerStatus->currentSpeed = 0.0f;
+        playerStatus->unk_8C = 0;
+        phi_f4 = -2.0f;
 
-    if (playerStatus->unk_0D != 0) {
-        if (gGameStatusPtr->peachFlags & 8) {
-            gGameStatusPtr->peachFlags &= ~0x8;
+        if (playerStatus->spriteFacingAngle >= 90.0f && playerStatus->spriteFacingAngle < 270.0f) {
+            phi_f4 = 2.0f;
+        }
+        *tempUnk_1C = phi_f4;
+        if ((playerStatus->animFlags & 0x2000) == 0) {
+            playerStatus->framesOnGround = 0x14;
+            playerStatus->fallState = 0;
+            parasolStruct->unk_08 = 0xF;
+            parasolStruct->npc = func_802B6000_E2A6B0();
         } else {
-            ret = npc_find_closest(playerStatus->position.x, playerStatus->position.y, playerStatus->position.z, 100.0f);
+            playerStatus->fallState = 0x14;
+            playerStatus->framesOnGround = 0x28;
+            parasolStruct->unk_04 = 1;
+            parasolStruct->unk_0C = 0xC;
+            temp_v0 = get_npc_by_index(D_8010C96C);
+            temp_v0->flags |= 0x40000;
+            playerStatus->flags |= 0x100000;
+            sfx_play_sound_at_player(0xFD, 0);
+        }
+    }
 
-            if (ret != NULL) {
-                if (fabs(ret->pos.y - playerStatus->position.y) - 1.0 > 0.0) {
-                    ret = NULL;
+    switch (playerStatus->fallState) {
+        case 0:
+            if (playerStatus->unk_90 == 0.0f) {
+                if (peach_disguise_check_overlaps() < 0) {
+                    suggest_player_anim_clearUnkFlag(0xC0024);
+                    sfx_play_sound_at_player(0x92, 0);
+                    playerStatus->fallState++;
                 } else {
-                    angle = clamp_angle(atan2(playerStatus->position.x, playerStatus->position.z, ret->pos.x, ret->pos.z));
-                    if (fabs(angle - func_800E5348()) > 30.0) {
-                         ret = NULL;
-                     }
+                    suggest_player_anim_clearUnkFlag(0xC0027);
+                    playerStatus->fallState = 0x32;
+                    playerStatus->framesOnGround = 0xA;
+                    parasolStruct->unk_08 = 0;
                 }
             }
-        }
+        case 1:
+            if (--playerStatus->framesOnGround == 0) {
+                suggest_player_anim_clearUnkFlag(0xC0025);
+                playerStatus->fallState = 2;
+                if (parasolStruct->npc == NULL) {
+                    playerStatus->fallState = 0x28;
+                } else {
+                    set_time_freeze_mode(TIME_FREEZE_FULL);
+                }
+            }
+            break;
+        case 2:
+            if (playerStatus->unk_BC != 0) {
+                playerStatus->framesOnGround = 0xC;
```

Comment:
Use decimal for everything but addresses and flags please 👉👈 
```suggestion
                playerStatus->framesOnGround = 12;
```

Reply:
Same goes for framesOnGround elsewhere in this func